gpkit/constraints/set.py

```diff
@@ -44,8 +38,11 @@ def __getitem__(self, key):
             if len(variables) == 1:
                 return variables[0]
             else:
-                variables.sort(key=_var_sort_key)
-                return variables
+                sortfn = lambda v: len(v.key.models) if v.key.models else 0
+                variables.sort(key=sortfn)
+                variable = variables[0]
+                # note: doesn't work for vector variables
```

Comment:
Probably they should be stored as arrays, not by their individual varkeys; but that's a issue which predates this and should have a new issue.